integration/kubernetes/k8s-ro-volume.bats

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2020 Ant Group
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../.ci/lib.sh"
+load "${BATS_TEST_DIRNAME}/../../lib/common.bash"
+
+setup() {
+	export KUBECONFIG="$HOME/.kube/config"
+	pod_name="test-readonly-volume"
+	container_name="busybox-ro-volume-container"
+	ro_volume_suffix="-tmp"
+	get_pod_config_dir
+}
+
+@test "Test readonly volume for pods" {
+	wait_time=20
+	sleep_time=2
+
+	# Create pod
+	kubectl create -f "${pod_config_dir}/pod-readonly-volume.yaml"
+
+	# Check pod creation
+	kubectl wait --for=condition=Ready pod "$pod_name"
+
+	# Validate readonly volume mount inside pod
+	check_cmd="mount|grep /tmp"
+	kubectl exec $pod_name -- sh -c "$check_cmd" | grep ro
+
+	# Validate readonly volume mount on the host
+	pod_id=$(sudo -E crictl pods -q -s Ready --name $pod_name)
+	container_id=$(sudo -E crictl ps -q --state Running --name $container_name)
+	shared_mounts="/run/kata-containers/shared/sandboxes/$pod_id/shared/"
+	host_mounts="/run/kata-containers/shared/sandboxes/$pod_id/mounts/"
+	mount | grep $shared_mounts | grep $container_id | grep -- $ro_volume_suffix | grep ro
```

Comment:
Could you use:

- `grep "/$container_id/"` to make matching the cid safer.
- `egrep '\<ro\>'` or `grep -w ro` to make the final `grep` safer.